### ISSUE TITLE
rtl8188ee issue on 4.15

### ISSUE THREAD

**Issue by myrage2000** [lwfinger/rtlwifi_new#327]
Hi,
A fix has been provided for rtl8188ee for Linux 4.15 on this github repo, but the 4.15 kernel is still missing this fix (https://github.com/torvalds/linux/tree/master/drivers/net/wireless/realtek/rtlwifi/rtl8188ee).
As a result, my realtek driver is not working at all for the moment.
What should I do to report this issue (or pull request) to a Linux maintainer please ?

I'm sorry for opening a issue for that, but I didn't find informations on how to do it.

**Comment by lwfinger** [comment]
You cannot request that a change applied to this repo be applied to the kernel. You must prepare and submit a patch according to the rules outlined in the Documentation directory of the kernel source tree.

What is the change that is in this repo, but not in the kernel? As far as I remember, all changes in this repo for rtl8188ee have been ported from mainline.

**Comment by myrage2000** [comment]
I will compare both repositories.
On 4.15 I cannot connect to a wpa2 router (it timeouts with no errors)
On 4.14 and before, everything works (with same versions of wpa_supplicant).
On ArchLinux and Gentoo forums, others (with rtl8188ee) have the same issue
(the driver is not working or the signal drops randomly).

I will contact them to do a report together, thanks for your help.

**Comment by lwfinger** [comment]
I did not have a 4.15 kernel available, but I tested with 4.15-rc7, which is almost the same. It connected to a WPA2 AP, and the connection seems solid. It has been up for about 45 minutes. I'm now building a 4.16-rc2 kernel for testing.

My distro is openSUSE 42.3, using NetworkManager.

**Comment by lwfinger** [comment]
On Leap 42.3, I used NetworkManager 1.0.12 and wpa_supplicant 2.2. It worked with kernel 4.15.3 and 4.16-rc2

I then booted openSUSE Tumbleweed using NetworkManager 1.10 and wpa_supplicant 2.6. It also worked with kernel 4.16.2.

I think there is something wrong with your user space. What distro and version are you using?

**Comment by myrage2000** [comment]
Thanks for your help.
I am using: ArchLinux (amd64) LiveUSB (so it's a clean environment - but I have the same issue on my desktop environment).
Linux 4.15.5; wpa_supplicant 2.6-11; networkmanager 1.10.5.
It does work if I downgrade the kernel to 4.14.21.

Here are some logs on 4.15.5:

> **lspci -k**
Network controller: Realtek Company RTL8188EE Wireless Network Adapter
Kernel driver in use: rtl8188ee
Kernel modules: rtl8188ee

> **dmesg | grep rtlwifi**
rtl8188ee: Using firmware rtlwifi/rtl8188efw.bin
rtlwifi: rtlwifi: wireless switch is on

Here I am using NetworkManager:
> **nmcli dev wifi connect _*[AP SSID]*_ password _*[AP PASSWORD]*_
Error: Connection activation failed: (5) IP configuration could not be reserved (no available address, timeout, etc.).

Next, I tried to manually connect to the AP using wpa_supplicant with wpa_cli:
> **wpa_cli**
scan
scan_results
-> _*[AP BSSID]*_ 2437 -54 [WPA-PSK-CCMP+TKIP][WPA2-PSK-CCMP+TKIP][WPS][ESS] _*[AP SSID]*_
add_network
-> 0
set_network 0 ssid _*[AP SSID]*_
-> OK
set_network 0 psk _*[AP PASSWORD]*_
-> OK
enable_network 0
-> OK
-> Trying to associate with _*[AP BSSID]*_
-> ...
-> CTRL-EVENT-CONNECTED - Connection to _*[AP BSSID]*_ completed [id=0 id_str=]

Then dhcpcd:
> **dhcpcd**
wlo1: soliciting an IPv6 router
wlo1: soliciting a DHCP lease
wlo1: no IPv6 Routers available
timed out

**Conclusion:** It looks like dhcpcd fails on 4.15, but works very well on 4.14. The interesting fact is that it happens with rtl8188ee for other people too.

Thanks again.

**Comment by lwfinger** [comment]
Until today, I had never used nmcli, and I have not used wpa_supplicant directly since the NetworkManager GUI became usable.

When I booted to a console (no X running), I was able to use nmcli to establish a connection with one of my APs without difficulty. It is a Netgear WNDR3400 running standard firmware. I did not try wpa_cli, but I have every indication that it would have worked.

As your system is the one with the problem, you are the one that will need to debug it. I suggest that you use git to clone the kernel sources from git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git and bisect between v4.15 (bad) and v4.14 (good) to find the commit that breaks ARCH. I suggest that you use the running configuration as a starting point for .config. It is likely stored in /boot/config-4.15xxxx.
